src/reducers/analyzeAreaSlice.js

```diff
@@ -30,11 +38,17 @@ export const AnalyzeAreaSlice = createSlice({
     changeGraphTableResilience: (state, action) => {
       state.isItAGraphResilience = !state.isItAGraphResilience;
     },
+    changeSortExpanded: (state, action) => {
+      state.isSortExpanded = !state.isSortExpanded;
+    },
     changeSortDirection: (state, action) => {
       state.isSortASC = !state.isSortASC;
     },
+    changeChartSortDirection: (state, action) => {
+      state.chartSortASC = !state.chartSortASC;
+    },
     changeSortBy: (state, action) => {
-      state.sortBy = action.payload;
+      state.sortBy = (action.payload !== state.sortBy) ? action.payload : null;
```

Comment:
I was observing some weird behavior with this... e.g. if you click on "Threat" and throw some logs of action.payload and state.sortBy in there - you can see that state.sortBy becomes null at some point.

I changed this to `state.sortBy = action.payload;` and it seems to work as expected. Just wanted to make sure there wasn't something I'm missing

Reply:
Oh I see. It's allowing you to remove the sorting.